test/logflare/lql/parser_test.exs

```diff
@@ -515,6 +534,79 @@ defmodule Logflare.LqlParserTest do
       assert err =~ "emailAddress"
       assert err =~ "metadata filter value"
     end
+
+    test "returns error when parser cannot handle part of input" do
+      assert {:error, "LQL parser doesn't know how to handle this part: " <> _rest} =
+               Parser.parse("event_message:test ][=invalid syntax", @default_schema)
+    end
+
+    test "returns error for general parsing errors" do
+      assert {:error, _err} = Parser.parse("\x00\x01\x02", @default_schema)
+    end
+
+    test "handles NimbleParsec 2-tuple error format" do
+      # trying to trigger this error with a null byte input
+      result = Parser.parse(IO.iodata_to_binary([<<0x00>>]), @default_schema)
+      assert match?({:error, _}, result)
+    end
+
+    test "returns error for map type fields" do
+      schema = build_schema(%{"metadata" => %{"config" => %{}}})
+
+      assert {:error, :field_not_found, "", ["Field type `map` is not queryable.", "", ""]} =
+               Parser.parse("m.config:value", schema)
+    end
+
+    test "handles naive_datetime type without casting" do
+      schema = build_schema(%{"created_at" => "2023-01-01T00:00:00"})
+
+      assert {:ok, [%FilterRule{path: "created_at", value: "test"}]} =
+               Parser.parse("created_at:test", schema)
+    end
+
+    test "handles string type without casting" do
+      schema = build_schema(%{"metadata" => %{"message" => "string_value"}})
+
+      assert {:ok, [%FilterRule{path: "metadata.message", value: "test_string"}]} =
+               Parser.parse("m.message:test_string", schema)
+    end
+
+    test "returns error for nonexistent field" do
+      assert {:error, :field_not_found, "",
+              [
+                "LQL parser error: path `metadata.nonexistent` not present in source schema.",
+                "",
+                ""
+              ]} =
+               Parser.parse("m.nonexistent:value", @default_schema)
+    end
+
+    test "handles naive_datetime fields without casting" do
+      schema = build_schema(%{"metadata" => %{"test_field" => "value"}})
+      mocked_typemap = %{"metadata.test_field" => :naive_datetime}
+
+      copy(Logflare.Google.BigQuery.SchemaUtils)
+
+      stub(Logflare.Google.BigQuery.SchemaUtils, :bq_schema_to_flat_typemap, fn _ ->
+        mocked_typemap
+      end)
+
+      assert {:ok, [%FilterRule{path: "metadata.test_field", value: "test_value"}]} =
+               Parser.parse("m.test_field:test_value", schema)
+    end
+
+    test "returns field not found error for nil type fields" do
+      schema = build_schema(%{"metadata" => %{"test_field" => "value"}})
+      mocked_typemap = %{"metadata.test_field" => nil}
+
+      copy(Logflare.Google.BigQuery.SchemaUtils)
+
+      stub(Logflare.Google.BigQuery.SchemaUtils, :bq_schema_to_flat_typemap, fn _ ->
+        mocked_typemap
+      end)
```

Comment:
we should be able to build a schema with `TestUtils.build_bq_schema/1`, we would be able to avoid stubs

Reply:
@nelsonmestevao